Gemfile.lock

```diff
@@ -0,0 +1,107 @@
+PATH
```

Comment:
Removed this file from the `.gitignore` so travis would pick up the gem upgrade

Reply:
This should stay `.gitignore`d. For libraries it's a best practice to keep `Gemfile.lock` out of your repo.

Reply:
yeah i just moved it back out. I think the issue was that we weren't using `bundle exec` for our `rubocop` command in the travis.yml

Reply:
We chatted in Slack, but wanted to add a less terse explanation here. So basically, when you are doing **application** development, `Gemfile.lock` helps a lot to make sure dependencies are consistent across both developers and deploys, minimizing "works on my machine" syndrome. But when you're developing a **library** that will be used in applications, you actually want the version constraints to be a little more flexible to accommodate varying dependencies in those applications. We don't want to pin a user to a specific version of Rails, for example, if we support versions `4.2` through `5.1`, and so forth. A longer discussion of the distinction can be found [here](http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/).

Adding `Gemfile.lock` here wouldn't actually cause problems with applications, since when a gem is installed it uses the `.gemspec` file to determine dependencies, not the `Gemfile.lock` in that gem. It does, however, impact developers on the project. If our gem claims to support different versions of its dependencies, then _it should not matter_ which of those versions are installed when you're developing the gem. And if it does cause problems when one of the dependencies pushes an update, you'd want to know about it so that you can either fix it or constrain things further. So, in a way, you don't check in `Gemfile.lock` to keep things a little more [stochastic](https://en.wikipedia.org/wiki/Stochastic).

Side note, I think the dependencies we have in our `Gemfile` really probably belong in `email_repair.gemspec` as development dependencies, also with relatively tight version constraints. Another distinction is between **development** and **runtime** dependencies for a gem. **development** dependencies are those that we use for developing the gem, such as `rake, `rubocop` and `rspec`. These should generally be more tightly constrained so that developers receive consistent versions. They are *not* installed when a user installs your gem. However, **runtime** dependencies are. These should be specified more loosely, within reason, to allow for our gem to be used in a wide variety of applications.

Reply:
thanks for the detailed explanation! super helpful

Reply:
@InformedK12/engineers if you're updating our non-app repos (like this one), ^ this is some good info on the difference between handling dependencies in application development (like synchroform) vs. library development (like this, junk drawer, etc)

Reply:
FYI I edited @mockdeep's comment to put it in the missing backtick cuz it was bothering me. Thanks for the explanation!

Reply:
Ha, so did I!